examples/platforms/simulation/radio.c

```diff
@@ -757,6 +770,25 @@ void platformRadioUpdateFdSet(fd_set *aReadFdSet, fd_set *aWriteFdSet, int *aMax
             *aMaxFd = sTxFd;
         }
     }
+
+    if (sEnergyScanning)
+    {
+        struct timeval tv  = {0, 0};
+        uint32_t       now = otPlatAlarmMilliGetNow();
+
+        if ((int32_t)(sEnergyScanEndTime - now) > 0)
```

Comment:
The cast from `uint` to `int` is not always safe (it is technically undefined behavior in C++ and up to toolchains). We can replace it with `uint` subtraction and comparison of result with `2^31`.